src/codec_aom.c

```diff
@@ -512,12 +512,18 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
 #endif
         int aomCpuUsed = -1;
         if (encoder->speed != AVIF_SPEED_DEFAULT) {
-            if (encoder->speed < 8) {
-                aomCpuUsed = AVIF_CLAMP(encoder->speed, 0, 6);
-            } else {
+            aomCpuUsed = AVIF_CLAMP(encoder->speed, 0, 9);
+#if defined(AOM_USAGE_ALL_INTRA) && \
+    AOM_ENCODER_ABI_VERSION >= (9 + AOM_CODEC_ABI_VERSION + AOM_EXT_PART_ABI_VERSION)
```

Comment:
There is a complication here: the AOM_EXT_PART_ABI_VERSION macro is new. If AOM_EXT_PART_ABI_VERSION is undefined, we are relying on the C preprocessor behavior that it replaces an undefined macro with 0L in expressions. Clang has a compiler warning for that. I suggest we do the following to avoid that Clang warning:


```
// Speeds 7-9 were added to all intra mode in https://aomedia-review.googlesource.com/c/aom/+/140624.
#if defined(AOM_EXT_PART_ABI_VERSION)
#if AOM_ENCODER_ABI_VERSION >= (10 + AOM_CODEC_ABI_VERSION + AOM_EXT_PART_ABI_VERSION)
#define ALL_INTRA_HAS_SPEEDS_7_TO_9 1
#endif
#endif
```

And then change lines 516-517 to

`#if defined(AOM_USAGE_ALL_INTRA) && defined(ALL_INTRA_HAS_SPEEDS_7_TO_9)`

Reply:
Done